From 6ccc322fe71f7d3f0eb8d4a77bf235dd814030f0 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 24 Oct 2023 16:41:43 +0200
Subject: [PATCH 407/841] feat(stm32mp2): enable DDR sub-system clock

Call ddr_sub_system_clk_init() just before clock driver init, as it
needs to be done before enabling DDR PLL clock (PLL2).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b40e02fbcfadaa55137a1e572b67720eb6be385b feat(stm32mp2): enable DDR sub-system clock]
Change-Id: I365d6aa034363d0c036ce2d9f944f077ba86e193
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/335916
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index 4e50ee042..8d79ce839 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -27,6 +27,7 @@
 #include <drivers/st/stm32mp_reset.h>
 #include <drivers/st/stm32mp_rifsc_regs.h>
 #include <drivers/st/stm32mp_risab_regs.h>
+#include <drivers/st/stm32mp2_ddr_helpers.h>
 #include <drivers/st/stm32mp2_ram.h>
 #include <drivers/st/stm32mp2_risaf.h>
 #include <lib/fconf/fconf.h>
@@ -237,6 +238,12 @@ void bl2_el3_plat_arch_setup(void)
 
 	reset_backup_domain();
 
+	/*
+	 * Initialize DDR sub-system clock. This needs to be done before enabling DDR PLL (PLL2),
+	 * and so before stm32mp2_clk_init().
+	 */
+	ddr_sub_system_clk_init();
+
 	if (stm32mp2_clk_init() < 0) {
 		panic();
 	}
-- 
2.47.2

